test/e2e/tests/test_alias.py

```diff
@@ -63,25 +81,107 @@ def simple_alias(simple_key):
+    def test_create_update_delete_alias(self, simple_alias, another_key):
         (ref, cr) = simple_alias
+        another_key_id = another_key['KeyMetadata']['KeyId']
 
+        assert k8s.wait_on_condition(ref, "ACK.ResourceSynced", "True", wait_periods=10)
         assert 'arn' in cr['status']['ackResourceMetadata']
         alias_arn = cr['status']['ackResourceMetadata']['arn']
 
-        _, deleted = k8s.delete_custom_resource(ref, 3, 5)
+        # Update
+        updates = {
+            "spec": {
+                "targetKeyID": another_key_id
+            }
+        }
+
+        k8s.patch_custom_resource(ref, updates)
+        time.sleep(MODIFY_WAIT_AFTER_SECONDS)
+        assert k8s.wait_on_condition(ref, "ACK.ResourceSynced", "True", wait_periods=10)
+        alias = kms_validator.get_alias(arn=alias_arn, target_key_id=another_key_id)
+        assert alias is not None, f"Alias should not be None for key id {another_key_id}"
+
+        _, deleted = k8s.delete_custom_resource(ref, DELETE_WAIT_PERIODS, DELETE_WAIT_PERIOD_LENGTH_SECONDS)
         assert deleted
+        time.sleep(DELETE_WAIT_AFTER_SECONDS)
+        kms_validator.assert_alias_deleted(arn=alias_arn, target_key_id=another_key_id)
+
+    def test_create_alias_ref(self):
+        key_name = random_suffix_name("ref-key", 15)
+        alias_name = random_suffix_name("ref-alias", 20)
+        replacements = REPLACEMENT_VALUES.copy()
+        replacements["REF_KEY_NAME"] = key_name
+        replacements["REF_ALIAS_NAME"] = alias_name
+
+        key_res_data = load_kms_resource(
+            "key_ref",
+            additional_replacements=replacements,
+        )
+
+        key_ref = k8s.CustomResourceReference(
+            CRD_GROUP, CRD_VERSION, KEY_RESOURCE_PLURAL,
+            key_name, namespace="default",
+        )
+
+        logging.debug(f"key resource. name: {key_name}, data: {key_res_data}")
+
+        alias_res_data = load_kms_resource(
+                "alias_ref",
+                additional_replacements=replacements,
+            )
+        logging.debug(alias_res_data)
+
+        # Create the k8s resource
+        alias_ref = k8s.CustomResourceReference(
+            CRD_GROUP, CRD_VERSION, ALIAS_RESOURCE_PLURAL,
+            alias_name, namespace="default",
+        )
+
+        logging.debug(f"alias resource. name: {alias_name}, data: {alias_res_data}")
+
+        # create the resources in order that initially the reference resolution fails and
+        # then when the referenced resource gets created, then all resolutions eventually
+        # pass and resources get synced.
+
+        # Create Alias. Needs Key reference
+        k8s.create_custom_resource(alias_ref, alias_res_data)
+
+        # Create Key. Needs no reference
+        k8s.create_custom_resource(key_ref, key_res_data)
+
+        time.sleep(CREATE_WAIT_AFTER_SECONDS)
+
+        assert k8s.wait_on_condition(key_ref, "ACK.ResourceSynced", "True", wait_periods=10)
```

Comment:
Replace the string const with an imported condition variable: https://github.com/aws-controllers-k8s/test-infra/blob/main/src/acktest/k8s/condition.py#L21